client/packages/purchasing/src/purchase_order/DetailView/LineEdit/utils.ts

```diff
@@ -86,3 +88,39 @@ export const calculatePricesAndDiscount = (
     }
   }
 };
+
+type UnitCalcFields = 'numberOfPacks' | 'packSize';
+
+export const calculateUnitQuantities = (
+  changingField: UnitCalcFields,
+  status: PurchaseOrderNodeStatus,
+  data: Partial<DraftPurchaseOrderLine>,
+  draft: DraftPurchaseOrderLine | null | undefined
+) => {
+  let numberOfPacks = draft?.numberOfPacks ?? 0;
+  let requestedPackSize = draft?.requestedPackSize ?? 0;
+
+  switch (changingField) {
+    case 'numberOfPacks': {
+      numberOfPacks = data.numberOfPacks || 0;
+      break;
+    }
+    case 'packSize': {
+      requestedPackSize = data.requestedPackSize || 0;
+      break;
+    }
+  }
```

Comment:
Don't think you need the switch here -- you're multiplying these two values together, it doesn't matter which one has changed in this case (it's only when there was three that it mattered)

Reply:
This is just the draft vs data thing I agree probably can remove it by the solution I mentioned above...